### PR TITLE
Msgformat macro

### PR DESCRIPTION
**Pull request by bazsi**
This PR adds the $MSGFORMAT name-value pair that contains the kind of parser we applied to parse the specific data. This is only set for syslog style messages (e.g. the ones using the msg-format mechanism), this helps you to diagnose misconfigured parsers.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: modules/syslogformat/syslog-format.c [hunk 1]
```diff
@@ -954,7 +954,7 @@ _syslog_format_parse_legacy(const MsgFormatOptions *parse_options,
     msg->flags |= LF_UTF8;
 
   log_msg_set_value(msg, LM_V_MESSAGE, (gchar *) src, left);
-
+  log_msg_set_value_to_string(msg, LM_V_MSGFORMAT, "rfc3164");
```
Comment (MrAnno) [inline modules/syslogformat/syslog-format.c:957]: A `return TRUE` path has been missed a few lines above.

File: modules/syslogformat/syslog-format.c [hunk 2]
```diff
@@ -1086,6 +1086,7 @@ _syslog_format_parse_syslog_proto(const MsgFormatOptions *parse_options, const g
         msg->flags |= LF_UTF8;
     }
   log_msg_set_value(msg, LM_V_MESSAGE, (gchar *) src, left);
+  log_msg_set_value_to_string(msg, LM_V_MSGFORMAT, "rfc5424");
```
Comment (MrAnno) [inline modules/syslogformat/syslog-format.c:1089]: Two `return TRUE` paths have been missed a few lines above.

**Review by MrAnno — DISMISSED** [review]

File: modules/syslogformat/syslog-format.c (hunk 1 above)
Comment (bazsi) [reply modules/syslogformat/syslog-format.c:1026]: I have fixed this by extracting the function that stores the message itself into a separate function. That should take care of this bug and make it more readable.

File: modules/syslogformat/syslog-format.c (hunk 2 above)
Comment (bazsi) [reply modules/syslogformat/syslog-format.c:1122]: same

**Review by MrAnno — APPROVED** [review]

**Pull request merged**